sources/iTermController.m

```diff
@@ -275,6 +275,34 @@ - (void)newWindow:(id)sender possiblyTmux:(BOOL)possiblyTmux {
     }
 }
 
+- (void)newWindowWithSameProfile:(id)sender possiblyTmux:(BOOL)possiblyTmux {
+    Profile *bookmark = nil;
+    if (_frontTerminalWindowController) {
```

Comment:
There's a bunch of duplicate code with the previous method and with `newSessionWithSameProfile`. Please refactor it out.

Reply:
Will do.